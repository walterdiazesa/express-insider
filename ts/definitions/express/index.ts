export enum HandlerType {
  ROUTE = "Route",
  MIDDLEWARE = "Middleware",
  ROUTE_MIDDLEWARE = "Route Middleware",
}

export * from './objects';
export * from './verbs';