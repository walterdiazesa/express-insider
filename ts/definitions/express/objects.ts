import { TrailOptions } from "..";
import { HandlerType, Method } from "../../../ts";

export type RouteMatcher = {
  route: TrailOptions['ignoreRoutes'][number]['route'],
  method: Lowercase<TrailOptions['ignoreRoutes'][number]['method']>,
};

export type Route = {
  path: `/${string}`;
  stack: RouteStack[];
  methods: Record<Method, boolean>;
};

type Stack<T extends HandlerType | undefined = undefined> = {
  handle: Function & { stack?: (StackItem | StackItem<HandlerType.ROUTE>)[] };
  name: string;
  params?: undefined;
  path: T extends HandlerType.ROUTE ? string : unknown;
  keys: Record<string, string | boolean | number>[];
  regexp: RegExp;
};

export type StackItem<T extends HandlerType | undefined = undefined> = Stack<T> & {
  route: T extends undefined ? undefined | Route : T extends HandlerType.ROUTE ? Route : undefined;
  // Config params
  ignore: T extends HandlerType.ROUTE ? undefined | true : never;
  showResponse: T extends HandlerType.ROUTE ? undefined | true : never;
  showRequestedURL: T extends HandlerType.ROUTE ? undefined | true : never;
}

export type RouteStack<T extends Method = Method> = Stack & {
  method: T;
};