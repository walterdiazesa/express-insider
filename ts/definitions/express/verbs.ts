export type Method = "get" | "post" | "options" | "all" | "patch" | "delete" | "put" | "head" | "connect" | "trace";