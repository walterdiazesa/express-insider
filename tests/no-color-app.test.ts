import request from 'supertest'
import express from 'express'
import { trail } from '../index';

export const noColorApp = express();

noColorApp.get('/', (req, res) => {
  res.send({ working: true })
})

noColorApp.all('/all-route', (req, res) => {
  res.status(304).send({ working: true })
})

trail(noColorApp, {
  trailId: () => 'test-id',
  timingFormatter: (elapsed) => 0,
  showColors: false
});

describe('trail with no color', () => {
  it('Should show handler message for GET / route', async () => {
    const log = jest.spyOn(global.console, 'log');
    await request(noColorApp).get("/");
    expect(log.mock.calls).toMatchSnapshot();
  })
  it('Should show handler message for [ALL] /all-route route', async () => {
    const log = jest.spyOn(global.console, 'log');
    await request(noColorApp).get("/all-route");
    expect(log.mock.calls).toMatchSnapshot();
  })
})