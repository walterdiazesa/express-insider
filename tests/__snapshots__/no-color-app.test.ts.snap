// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trail with no color Should show handler message for [ALL] /all-route route 1`] = `
[
  [
    "[test-id] GET /all-route start",
  ],
  [
    "[test-id] Middleware query 0 ms",
  ],
  [
    "[test-id] Middleware expressInit 0 ms",
  ],
  [
    "[test-id] Route GET /all-route",
  ],
  [
    "[test-id]                     <anonymous (6)>",
  ],
  [
    "[test-id]                                    (send response) 0 ms",
  ],
  [
    "[test-id]                                    304 Not Modified, total elapsed time: 0 ms",
  ],
  [
    "[test-id]                                    (cleanup) 0 ms",
  ],
  [
    "[test-id]                                    (total) 0 ms",
  ],
  [
    "[test-id] GET /all-route finish, elapsed time since begin: 0 ms",
  ],
]
`;

exports[`trail with no color Should show handler message for GET / route 1`] = `
[
  [
    "[test-id] GET / start",
  ],
  [
    "[test-id] Middleware query 0 ms",
  ],
  [
    "[test-id] Middleware expressInit 0 ms",
  ],
  [
    "[test-id] Route GET / 0 ms 200 OK",
  ],
  [
    "[test-id] GET / finish, elapsed time since begin: 0 ms",
  ],
]
`;
